docs/rules/no-hooks-from-ancestor-modules.md

```diff
@@ -0,0 +1,43 @@
+# Disallow the use of hooks from ancestor modules (no-hooks-from-ancestor-modules)
+
+When a QUnit `module` is used with a nested callback, the callback provides a `hooks`
+object as its first argument. This allows calling `hooks.beforeEach` and `hooks.afterEach`
+within that callback's body.
+
+More deeply nested `module` uses should use their own callback's provided `hooks` argument
+and not one from an ancestor `module`.
+
+## Rule Details
+
+The following patterns are considered warnings:
+
+```js
+QUnit.module("outer module", function(hooks) {
+  QUnit.module("inner module", function() {
+    hooks.beforeEach(function() {});
+  });
+});
+
+QUnit.module("outer module", function(outerHooks) {
+  QUnit.module("inner module", function() {
```

Comment:
This example looks pretty much the same as the example above. Let me know if I'm missing something, but otherwise I think it makes sense to show an inner variable:

```suggestion
  QUnit.module("inner module", function(innerHooks) {
```